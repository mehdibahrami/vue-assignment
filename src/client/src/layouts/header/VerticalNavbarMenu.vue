<template>
  <div v-if="isOpen" class="navbar-vertical-menu">
    <!-- Menu Items -->
    <div
      v-for="(item, index) in items"
      :key="index"
      class="navbar-vertical-items"
    >
      <a :href="item.url" class="navbar-vertical-selected-button">
        {{ item.title }}
      </a>
    </div>
  </div>
</template>

<script>
export default {
  name: "VerticalNavbarMenu",
  props: {
    isOpen: {
      type: Boolean,
      default: false,
    },
    items: { type: Array, required: true },
  },
};
</script>

<style scoped lang="scss">
.navbar-vertical-menu {
  border-bottom-width: 1px;
  border-color: $green-dark;
}

.navbar-vertical-items {
  padding: 0.75rem 0.5rem 0.75rem 0.5rem;
}

.navbar-vertical-selected-button {
  background-color: $green-dark;
  color: $white;
  display: block;
  padding: 0.5rem 0.75rem 0.5rem 0.75rem;
  border-radius: 0.375rem;
  font-size: $font-size-base;
}

@media (min-width: 640px) {
  .navbar-vertical-items {
    padding-left: 0.75rem;
    padding-right: 0.75rem;
  }
}

@media (min-width: 768px) {
  .navbar-vertical-menu {
    display: none;
  }
}
</style>
