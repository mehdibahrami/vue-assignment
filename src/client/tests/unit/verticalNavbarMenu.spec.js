import { shallowMount } from "@vue/test-utils";
import VerticalNavbarMenu from "@/layouts/header/VerticalNavbarMenu.vue";

describe("VerticalNavbarMenu", () => {
  let wrapper;

  beforeEach(() => {
    // shallow mounting the main component
    wrapper = shallowMount(VerticalNavbarMenu, {
      props: {
        isOpen: true,
        items: [
          {
            url: "/",
            title: "Dashboard",
          },
        ],
      },
    });
  });

  it("renders correctly", () => {
    expect(
      wrapper.find("a.navbar-vertical-selected-button").attributes(["href"])
    ).toBe("/");
    expect(wrapper.find("a.navbar-vertical-selected-button").text()).toBe(
      "Dashboard"
    );
    expect(wrapper.element).toMatchSnapshot();
  });

  it("should be displayed when isOpen prop is true", () => {
    expect(wrapper.find(".navbar-vertical-menu").exists()).toBeTruthy();
  });

  it("should not be displayed when isOpen prop is false", async () => {
    await wrapper.setProps({ isOpen: false });

    expect(wrapper.find(".navbar-vertical-menu").exists()).toBe(false);
  });
});
