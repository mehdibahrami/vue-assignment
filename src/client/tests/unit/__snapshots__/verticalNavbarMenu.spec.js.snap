// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalNavbarMenu renders correctly 1`] = `
<div
  class="navbar-vertical-menu"
>
  <!-- Menu Items -->
  
  <div
    class="navbar-vertical-items"
  >
    <a
      class="navbar-vertical-selected-button"
      href="/"
    >
      Dashboard
    </a>
  </div>
  
</div>
`;
